go/obscuronode/enclave/sql/edbattestation.go

```diff
@@ -0,0 +1,204 @@
+	}
+	certs = append(certs, block)
+
+	// If we get more than one certificate, append it to the slice
+	for len(rest) > 0 {
+		block, rest = pem.Decode(rest)
+		if block == nil {
+			return nil, tcbstatus.Unknown, errors.New("could not parse certificate chain")
+		}
+		certs = append(certs, block)
+	}
+
+	if verifyRemoteReport == nil {
+		return certs, tcbstatus.Unknown, nil
+	}
+
+	if len(quote) == 0 {
+		return nil, tcbstatus.Unknown, ErrEmptyQuote
+	}
+
+	report, verifyErr := verifyRemoteReport(quote)
+	if verifyErr != nil && !errors.Is(verifyErr, attestation.ErrTCBLevelInvalid) {
+		return nil, tcbstatus.Unknown, verifyErr
+	}
+
+	// Use Root CA (last entry in certs) for attestation
+	certRaw := certs[len(certs)-1].Bytes
+
+	if err := verifyReport(report, certRaw, config); err != nil {
+		return nil, tcbstatus.Unknown, err
+	}
+
+	return certs, report.TCBStatus, verifyErr
+}
+
+func verifyReport(report attestation.Report, cert []byte, config []byte) error {
+	hash := sha256.Sum256(cert)
+	if !bytes.Equal(report.Data[:len(hash)], hash[:]) {
+		return errors.New("report data does not match the certificate's hash")
+	}
+
+	var cfg struct {
+		SecurityVersion uint
+		UniqueID        string
+		SignerID        string
+		ProductID       uint16
+		Debug           bool
+	}
+	if err := json.Unmarshal(config, &cfg); err != nil {
+		return err
+	}
+	if cfg.UniqueID == "" {
+		if cfg.SecurityVersion == 0 {
+			return errors.New("missing securityVersion in config")
+		}
+		if cfg.ProductID == 0 {
+			return errors.New("missing productID in config")
+		}
+	}
+
+	if cfg.SecurityVersion != 0 && report.SecurityVersion < cfg.SecurityVersion {
+		return errors.New("invalid security version")
+	}
+	if cfg.ProductID != 0 && binary.LittleEndian.Uint16(report.ProductID) != cfg.ProductID {
+		return errors.New("invalid product")
+	}
+	if report.Debug && !cfg.Debug {
+		return errors.New("debug enclave not allowed")
+	}
+	if err := verifyID(cfg.UniqueID, report.UniqueID, "unqiueID"); err != nil {
+		return err
+	}
+	if err := verifyID(cfg.SignerID, report.SignerID, "signerID"); err != nil {
+		return err
+	}
+	if cfg.UniqueID == "" && cfg.SignerID == "" {
+		fmt.Println("Warning: Configuration contains neither uniqueID nor signerID!")
+	}
+
+	return nil
+}
+
+func verifyID(expected string, actual []byte, name string) error {
+	if expected == "" {
+		return nil
+	}
+	expectedBytes, err := hex.DecodeString(expected)
+	if err != nil {
+		return err
+	}
+	if !bytes.Equal(expectedBytes, actual) {
+		return errors.New("invalid " + name)
+	}
+	return nil
+}
+
+func httpGetCertQuote(tlsConfig *tls.Config, host, path string) (string, []byte, error) {
+	client := http.Client{Transport: &http.Transport{TLSClientConfig: tlsConfig}}
+	url := url.URL{Scheme: "https", Host: host, Path: path}
+	resp, err := client.Get(url.String()) //nolint:noctx
+	if err != nil {
+		return "", nil, err
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	var certquote certQuoteResp
```

Comment:
@BedrockSquirrel Not sure what this variable declaration is doing all the way up here. Move it down :)